stage #2.5/module #3 Spring Core/#1. Spring pros and cons/content.md

```diff
@@ -0,0 +1,50 @@
+   wide-ranging set of features and functionality. It makes the job simpler.
+3. **Loose Coupling**: Spring applications are loosely coupled because of dependency injection. It handles injecting
+   dependent components without a component knowing where they came from.
+4. **Powerful Abstraction**: It provides a powerful abstraction to JEE specifications such as JMS, JDBC, JPA, and JTA.
+5. **Declarative Support**: It provides declarative support for caching, validation, transaction, and formatting.
+6. **Portable**: We can use server-side in web/EJB app, client-side in swing app business logic is completely portable.
+7. **Cross-cutting behavior**: Resource management is a cross-cutting concern, easy to copy and paste everywhere.
+8. **Configuration**: It provides a consistent way of configuring everything, separate configuration from application
+   logic, varying configuration.
+9. **Lifecycle**: Responsible for managing all your application components, particularly those in middle-tier container
+   sees components through well-defined lifecycle: init(), destroy().
+10. **Dependency Injection**: The use of dependency injection makes the easy development of JavaEE.
+11. **Easier Testing**: The use of dependency injection makes the testing easy. The spring framework does not require a
+    server while the EJB and Struts application requires a server.
+12. **Fast**: The team of Spring engineers deeply cares about the performance. Its fast startup, fast shutdown, and
+    optimized execution maintain performance make it fast. Even, we can start a new Spring project in seconds by using
+    Spring Initializr.
+13. **Secure**: It monitors third-party dependencies closely. The regular update is issued that make our data and
+    application secure. We can make our application secure by using the Spring Security framework. It provides
+    industry-standard security schemes and delivers a trustworthy solution that is secure by default.
+14. **Supportive**: The Spring community provides support and resources to get you to the next level QuickStart guides,
+    tutorials, videos, and meetup helps a lot.
+15. **Productive**: It is more productive because the spring application can integrate with other Spring-based
+    applications. For example, we can combine the Spring Boot application with Spring Cloud.
+
+While there are advantages of Spring framework, there are some disadvantages as well, and they are listed below:
+
+1. **Complexity**: Working with Spring is more complex. It requires a lot of expertise. If you have not used Spring
+   before, first you will have to learn. The learning curve is also difficult, so if you have not a lot of development
+   experience, it is difficult to learn.
+2. **Parallel Mechanism**: It provides multiple options to developers. These options create confusion to developers that
+   which feature to use and which to not and wrong decisions may lead to significant delays.
```

Comment:
the sentence is broken, the meaning is not clear

Reply:
adjusted, thanks